src/main/java/com/google/devtools/build/lib/authandtls/Netrc.java

```diff
@@ -0,0 +1,87 @@
+// Copyright 2020 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package com.google.devtools.build.lib.authandtls;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableMap;
+import java.io.IOException;
+import java.io.InputStream;
+import javax.annotation.Nullable;
+
+public class Netrc {
+
+  @Nullable
+  private final Credential defaultCredential;
+  private final ImmutableMap<String, Credential> credentials;
+
+  public static Netrc fromStream(InputStream inputStream) throws IOException  {
+    return NetrcParser.parseAndClose(inputStream);
+  }
+
+  public Netrc(@Nullable Credential defaultCredential, ImmutableMap<String, Credential> credentials) {
+    this.defaultCredential = defaultCredential;
+    this.credentials = credentials;
+  }
+
+  @Nullable
+  public Credential getCredential(String machine) {
+    return credentials.getOrDefault(machine, defaultCredential);
+  }
+
+  @VisibleForTesting
+  @Nullable
+  Credential getDefaultCredential() {
+    return defaultCredential;
+  }
+
+  @VisibleForTesting
+  ImmutableMap<String, Credential> getCredentials() {
+    return credentials;
+  }
+
+  @AutoValue
+  public static abstract class Credential {
```

Comment:
Does `@AutoValue` produce `toString` that would leak the password?
If so there should be an override for it that would display as `<password>` or something.

Reply:
Updated.